sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -36,13 +35,15 @@ class CloudEvent(EventMixin):   #pylint:disable=too-many-instance-attributes
     """Properties of an event published to an Event Grid topic using the CloudEvent 1.0 Schema.
 
     All required parameters must be populated in order to send to Azure.
+    If data is of binary type, data_base64 can be used alternatively. Note that data and data_base64
+    cannot be present at the same time.
 
     :param source: Required. Identifies the context in which an event happened. The combination of id and source must
         be unique for each distinct event. If publishing to a domain topic, source must be the domain name.
     :type source: str
     :param type: Required. Type of event related to the originating occurrence.
     :type type: str
-    :param data: Required. Event data specific to the event type.
+    :keyword data: Optional. Event data specific to the event type. Only one of the `data` or `data_base64` argument must be present.
```

Comment:
as OOB discussion:
we could add a note saying that `if data is of bytes type, we'll send it as `data_base64` in the outgoing request`

By this we add clarity on what users should expect on the data of outgoing request e.g. request.get("data") is None, request.get("data_base64") has value.